crypto/initthread.c

```diff
@@ -281,15 +283,15 @@ static const OSSL_LIB_CTX_METHOD thread_event_ossl_ctx_method = {
 void ossl_ctx_thread_stop(void *arg)
 {
     THREAD_EVENT_HANDLER **hands;
-    OSSL_LIB_CTX *ctx = arg;
+    OSSL_LIB_CTX *ctx = PROV_LIBCTX_OF(arg);
```

Comment:
Side note: Frankly, it's really really really confusing to find this level of provider specific code in `crypto/`...  It took me a moment to recollect myself after an initial knee-jerk reaction where I wanted to scream "NO" at the top of my lungs.